src/main/java/seedu/address/model/animal/TaskList.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.model.animal;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the TaskList containing tasks that are addressed to the {@link Animal}
+ *
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructs a TaskList that stores Tasks.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Returns the taskList.
+     * @return the taskList containing tasks.
+     */
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+    /**
+     * Returns all tasks in the TaskList as a string, separated by a newline
+     * @return A string of tasks, separated by a newline
+     */
+    public String getTasksAsString() {
+        return taskList.stream()
+                .map(Task::toString)
+                .collect(Collectors.joining("\n"));
```

Comment:
Is there a reason why you did not put this in TaskList::toString instead?

Reply:
Actually, you brought up a great point, I will change that. Thanks!